e2e/README.md

```diff
@@ -100,6 +100,84 @@ In order to debug the `godog` tests, 1) you must have the `runner_test.go` file
 
 ![](./debug.png)
 
+## Regression testing
+We have built the project and the CI job in a manner that it is possible to override different parameters about projects versions, so that we can set i.e. the version of the Elastic Stack to be used, or the version of the Elastic Agent. There also exist maintenance branches where we set the specific versions used for the tests:
+
+- **7.9.x**: will use `7.9.x` alias for the Elastic Stack, the Agent and Metricbeat
+- **7.10.x**: will use `7.10.x` alias for the Elastic Stack, the Agent and Metricbeat
+- **7.x**: will use `7.x` alias for the Elastic Stack, the Agent and Metricbeat
+- **master**: will use `8.0.0-SNAPSHOT` for the Elastic Stack, the Agent and Metricbeat, representing the current development version of the different products under test.
+
+With that in mind, the project supports setting these versions in environment variables, overriding the pre-branch default ones.
+
+### Overriding Product versions
+We are going to enumerate the variables that will affect the product versions used in the tests, per test suite:
+
+>It's important to notice that the 7.9.x branch in **Fleet** test suite uses different source code for the communications with Kibana Fleet plugin, as API endpoints changed from 7.9 to 7.10, so there could be some combinations that are broken. See https://github.com/elastic/e2e-testing/pull/348 for further reference about these breaking changes.
+
+> Related to this compatibility matrix too, it's also remarkable that Kibana **Fleet** plugin should not allow to enroll an agent with a version higher than kibana (See https://github.com/elastic/kibana/blob/fed9a4fddcc0087ee9eca6582a2a84e001890f08/x-pack/test/fleet_api_integration/apis/agents/enroll.ts#L99).
+
+#### Fleet
+- `ELASTIC_AGENT_VERSION`. Set this environment variable to the proper version of the Elastic Agent to be used in the current execution. Default: See https://github.com/elastic/e2e-testing/blob/master/.ci/Jenkinsfile#L36
```

Comment:
NIT:

what do you think if the link to the L36 is using a specific commit instead? Just wondering if the pipeline gets changes this particular link will not point to the expected line anymore

Reply:
Sure, let me update! Thanks for the feedback